helpdesk/templates/helpdesk/ticket.html

```diff
@@ -109,39 +109,23 @@ <h5 class="mb-1">{{ followup.title|num_to_link }}</h5>
 
                 <dt><label>{% trans "New Status" %}</label></dt>
                 {% if not ticket.can_be_resolved %}<dd>{% trans "This ticket cannot be resolved or closed until the tickets it depends on are resolved." %}</dd>{% endif %}
-                {% if ticket.status == 1 %}
                 <dd><div class="form-group">
-                    <label for='st_open' class='active radio-inline'><input type='radio' name='new_status' value='1' id='st_open' checked='checked'>{% trans "Open" %} &raquo;</label>
-                    <label for='st_resolved' class="radio-inline"><input type='radio' name='new_status' value='3' id='st_resolved'{% if not ticket.can_be_resolved %} disabled='disabled'{% endif %}>{% trans "Resolved" %} &raquo;</label>
-                    <label for='st_closed' class="radio-inline"><input type='radio' name='new_status' value='4' id='st_closed'{% if not ticket.can_be_resolved %} disabled='disabled'{% endif %}>{% trans "Closed" %} &raquo;</label>
-                    <label class="radio-inline" for='st_duplicate'><input type='radio' name='new_status' value='5' id='st_duplicate'>{% trans "Duplicate" %}</label>
-                </div></dd>
+                {% for status_choice in ticket.STATUS_CHOICES %}
+                {% with status_choice_strfmt_s=status_choice.0|stringformat:"s" %}
+                {% if ticket.status == 1 and status_choice_strfmt_s in "2" %}{% else %}
+                {% if ticket.status == 2 and status_choice_strfmt_s in "1" %}{% else %}
+                {% if ticket.status == 3 and status_choice_strfmt_s not in "2,3,4" %}{% else %}
+                {% if ticket.status == 4 and status_choice_strfmt_s not in "2,4" %}{% else %}
+                {% if ticket.status == 5 and status_choice_strfmt_s not in "2,5" %}{% else %}
```

Comment:
Nice refactoring already! Maybe creating a custom template tag could help simplifying this big if.

We would have something like :  
```suggestion
                {% should_display_status ticket.status status_choice as should_display_status_button %}
                {% if should_display_status_button %}
```

For the implementation of the template tag, I imagine something like this :  
```python
@register.simple_tag
def should_display_status(ticket_status, status_choice):
    # Create a mapping between current status and possible next statuses
    mapping = {
        1: [2],
        2: [1],
        3: [2, 3, 4],
        4: [2, 4],
        5: [2, 5],
    }
    return status_choice in mapping[ticket_status]
```

Maybe, we could add a third parameter in order to reuse it in the public view ticket which seem to have lightly different mapping rules.